parking/backend/engine/alloc_engine.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from typing import Optional
+from geopy.distance import distance
+from parking.backend.db.dbaccess import DbAccess
+from parking.shared.rest_models import ParkingLot
+from parking.shared.location import Location
+from parking.shared.ws_models import ParkingRequestMessage
+
+logger = logging.getLogger('backend')
+
+
+class AllocationEngine():
+    def __init__(self, dba: DbAccess, user_sessions: 'UserSessions') -> None:
+        self.dba = dba
+        self.user_sessions = user_sessions
+
+    async def handle_allocation_request(self, user_id: str, request: ParkingRequestMessage) -> Optional[ParkingLot]:
+        rejections = self.user_sessions.get_user(user_id).rejections
+        lots = await self.dba.get_available_parking_lots(request.location, 100, rejections)
+
+        # Just take the first lot for now...
+        if lots:
+            lot = lots[0]
+            return ParkingLot(lot['capacity'], lot['name'], lot['price'], Location(lot['lat'], lot['long']))
```

Comment:
It would be a lot nicer if you could just pass `lot` into the constructor and get an object, rather than passing each array element in... A quick google brings up classmethods, what are your thoughts on this? 
https://stackoverflow.com/questions/141545/how-to-overload-init-method-based-on-argument-type

Reply:
It'd be nice to have a better way of constructing the object, a classmethod could be a good idea. I'll think about it more tomorrow and will have something better for the final PR